src/Database/Type/BoolType.php

```diff
@@ -24,10 +24,40 @@
  *
  * Use to convert bool data between PHP and the database types.
  */
-class BoolType extends Type
+class BoolType implements TypeInterface
 {
 
     /**
+     * Identifier name for this type
+     *
+     * @var string
+     */
+    protected $_name = null;
+
+    /**
+     * Constructor
+     *
+     * @param string|null $name The name identifying this type
+     */
+    public function __construct($name = null)
```

Comment:
Removing the parent class could break userland typehints. I think that's probably pretty rare though, and is an acceptable level of break for me.